### ISSUE TITLE
Extern builds should reflect the top-level binary directory

### ISSUE THREAD

**Issue by program--** [NOAA-OWP/ngen#732]
Currently, when `NGEN_WITH_EXTERN_*` is `ON`, the external dependencies are *always* built into `<extern_dir>/<extern_dep>/cmake_build`. Instead of `cmake_build`, the subdirectory should reflect the name of the top-level directory, OR they should be placed into the top-level directory.

This is probably better handling in the event configurations change compiler flags or some other CMake setting, but we'd want to keep previous versions. i.e. I build NGen into `build-debug` and `build-release`; in both cases, there will only be one build directory for the models, `cmake_build`. Ideally, it should be `<model_dir>/build-debug` and `<model_dir>/build-release`, or similar.

## Current behavior

Extern models always build into `cmake_build`.

## Expected behavior

Extern models binary dirs should reflect the top-level binary dir, or build directly into it.

## Steps to replicate behavior (include URLs)

1. clone NGen
2. Build into a non-`cmake_build` directory.
3. Build into a separate, distinct non-`cmake_build` directory.
4. Observe that extern models built only are built into the single `cmake_build` directory.

**Comment by PhilMiller** [comment]
I'd honestly like to see the extern stuff built beneath the binary directory, and not anywhere relative to the source tree.

**Comment by program--** [comment]
> I'd honestly like to see the extern stuff built beneath the binary directory, and not anywhere relative to the source tree.

Yeah this is the solution I'm leaning toward, as it aligns with best practices and feels the most stable.

**Comment by stcui007** [comment]
You mean in the cmake_build/?

**Comment by stcui007** [comment]
You mean in cmake_build?

**Comment by program--** [comment]
@stcui007 Yes, or whatever build/binary directory the user specifies.

**Comment by stcui007** [comment]
Don't you have to build a new library set every time you build a new binary?

**Comment by PhilMiller** [comment]
> Don't you have to build a new library set every time you build a new binary?

Yes, that's the point. If one's binary is configured differently, the automatically-built libraries should be configured to match.

**Comment by stcui007** [comment]
For automatic build process, That sounds like a good idea..
